src/types/check_config.rs

```diff
@@ -328,99 +339,154 @@ mod tests {
+        );
+        run_interval_test(
+            &multi_region_config,
+            TestInterval::OneMinute,
+            2,
+            TestInterval::OneSecond,
+            1,
+            vec!["eu_west".to_string()],
+            vec!["us_west".to_string(), "us_east".to_string()],
+        );
+        run_interval_test(
+            &multi_region_config,
+            TestInterval::OneMinute,
+            3,
+            TestInterval::OneSecond,
+            1,
+            vec!["us_west".to_string()],
+            vec!["eu_west".to_string(), "us_east".to_string()],
+        );
 
         // Try different spots in the minute
-        assert!(multi_region_config.should_run(
-            Tick::from_time(DateTime::from_timestamp(29, 0).unwrap()),
-            &"us_west".to_string()
-        ));
-        assert!(multi_region_config.should_run(
-            Tick::from_time(DateTime::from_timestamp(59, 0).unwrap()),
-            &"us_west".to_string()
-        ));
-        assert!(!multi_region_config.should_run(
-            Tick::from_time(DateTime::from_timestamp(83, 0).unwrap()),
-            &"us_west".to_string()
-        ));
-        assert!(multi_region_config.should_run(
-            Tick::from_time(DateTime::from_timestamp(83, 0).unwrap()),
-            &"us_east".to_string()
-        ));
-        assert!(!multi_region_config.should_run(
-            Tick::from_time(DateTime::from_timestamp(119, 0).unwrap()),
-            &"us_west".to_string()
-        ));
-        assert!(multi_region_config.should_run(
-            Tick::from_time(DateTime::from_timestamp(119, 0).unwrap()),
-            &"us_east".to_string()
-        ));
-        assert!(!multi_region_config.should_run(
-            Tick::from_time(DateTime::from_timestamp(145, 0).unwrap()),
-            &"us_west".to_string()
-        ));
-        assert!(multi_region_config.should_run(
-            Tick::from_time(DateTime::from_timestamp(145, 0).unwrap()),
-            &"eu_west".to_string()
-        ));
-        assert!(!multi_region_config.should_run(
-            Tick::from_time(DateTime::from_timestamp(171, 0).unwrap()),
-            &"us_west".to_string()
-        ));
-        assert!(multi_region_config.should_run(
-            Tick::from_time(DateTime::from_timestamp(171, 0).unwrap()),
-            &"eu_west".to_string()
-        ));
-        assert!(multi_region_config.should_run(
-            Tick::from_time(DateTime::from_timestamp(220, 0).unwrap()),
-            &"us_west".to_string()
-        ));
-        assert!(multi_region_config.should_run(
-            Tick::from_time(DateTime::from_timestamp(200, 0).unwrap()),
-            &"us_west".to_string()
-        ));
+        run_interval_test(
+            &multi_region_config,
+            TestInterval::OneMinute,
+            0,
+            TestInterval::OneSecond,
+            29,
+            vec!["us_west".to_string()],
+            vec!["eu_west".to_string(), "us_east".to_string()],
+        );
```

Comment:
pretty small nit, but I think I do kind of wish it took a struct so we could see what the names of each of these parameters are. but I'm sure some IDEs show ghost labels for the parameters.

indifferent